packages/cardano-services/test/Asset/CardanoTokenRegistry.test.ts

```diff
@@ -0,0 +1,74 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { CardanoTokenRegistry } from '../../src/Asset';
+import { InMemoryCache, Key } from '../../src/InMemoryCache';
+import { dummyLogger } from 'ts-log';
+
+describe('CardanoTokenRegistry', () => {
+  const nonvalidAssetId = Cardano.AssetId('0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef');
+  const validAssetId = Cardano.AssetId('f43a62fdc3965df486de8a0d32fe800963589c41b38946602a0dc53541474958');
+
+  describe('return value', () => {
+    const tokenRegistry = new CardanoTokenRegistry({ logger: dummyLogger });
+
+    afterAll(() => tokenRegistry.shutdown());
+
+    it('returns null for nonexisting AssetId', async () => {
+      expect(await tokenRegistry.getTokenMetadata([nonvalidAssetId])).toEqual([null]);
```

Comment:
maybe it's better to have just empty list?

Reply:
The value is there to cover partial match for the complete query, e.g `[null, { subject: 'abc', ... }]`

Reply:
Not only. I chosen this result type also to be immediately compliant with [this comment](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Asset/types/AssetInfo.ts#L26).